src/main/java/sorald/cli/Cli.java

```diff
@@ -171,11 +181,69 @@ private void validateArgs() {
                         RepairStrategy.SEGMENT.name()
                                 + " repair does not currently support statistics collection");
             }
+
+            validateRuleKeys();
+        }
+
+        /** Perform further processing of raw command line args. */
+        private void postprocessArgs() {
+            ruleViolations = parseRuleViolations(rules);
+            ruleKeys = parseRuleKeys(rules, ruleViolations);
+        }
+
+        private List<RuleViolation> parseRuleViolations(Rules rules) {
+            return rules.ruleViolationSpecifiers.stream()
+                    .map(this::parseRuleViolation)
+                    .collect(Collectors.toUnmodifiableList());
+        }
+
+        private List<Integer> parseRuleKeys(Rules rules, List<RuleViolation> ruleViolations) {
+            return ruleViolations.isEmpty()
+                    ? rules.ruleKeys
+                    : ruleViolations.stream()
+                            .map(RuleViolation::getRuleKey)
+                            .map(Integer::parseInt)
+                            .collect(Collectors.toUnmodifiableList());
+        }
+
+        private void validateRuleKeys() {
+            for (Integer ruleKey : ruleKeys) {
+                if (Processors.getProcessor(ruleKey) == null) {
+                    throw new CommandLine.ParameterException(
+                            spec.commandLine(),
+                            "Sorry, repair not available for rule "
+                                    + ruleKey
+                                    + ". See the available rules below.");
+                }
+            }
+        }
+
+        private RuleViolation parseRuleViolation(String violationSpecifier) {
+            String[] parts = violationSpecifier.split(Constants.VIOLATION_SPECIFIER_SEP);
+            String key = parts[0];
+            String rawFilename = parts[1];
+            String fileName =
+                    originalFilesPath.toPath().resolve(rawFilename).toAbsolutePath().toString();
+
+            if (!new File(fileName).isFile()) {
+                throw new CommandLine.ParameterException(
+                        spec.commandLine(),
+                        String.format(
+                                "Invalid violation ID '%s', no file '%s' in directory '%s'",
+                                violationSpecifier, rawFilename, originalFilesPath));
+            }
+
+            int startLine = Integer.parseInt(parts[2]);
+            int startCol = Integer.parseInt(parts[3]);
+            int endLine = Integer.parseInt(parts[4]);
+            int endCol = Integer.parseInt(parts[5]);
+            return new SpecifiedViolation(key, fileName, startLine, endLine, startCol, endCol);
```

Comment:
I intend to refactor all of this specifier parsing into a separate class, but that'll be for a future PR.

Reply:
It can currently crash not-so-gracefully on poorly formatted specifiers (e.g. incorrect amount of elements in the specifier, numbers not actually being numbers, etc)